qcodes/instrument_drivers/Keysight/Infiniium.py

```diff
@@ -152,6 +150,122 @@ def get(self):
+                           label='Measurement secondary source',
+                           set_cmd=partial(self._set_source, 2),
+                           get_cmd=partial(self._get_source, 2),
+                           val_mapping={i: f'CHAN{i}' for i in range(1, 5)})
+
+        self.add_parameter(name='amplitude',
+                           label='Voltage amplitude',
+                           get_cmd=self._meas('VAMPlitude'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='average',
+                           label='Voltage average',
+                           get_cmd=self._meas('VAVerage'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='base',
+                           label='Statistical base',
+                           get_cmd=self._meas('VBASe'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='lower',
+                           label='Voltage lower',
+                           get_cmd=self._meas('VLOWer'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='max',
+                           label='Voltage maximum',
+                           get_cmd=self._meas('VMAX'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='middle',
+                           label='Middle threshold voltage',
+                           get_cmd=self._meas('VMIDdle'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='min',
+                           label='Voltage minimum',
+                           get_cmd=self._meas('VMIN'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='overshoot',
+                           label='Voltage overshoot',
+                           get_cmd=self._meas('VOVershoot'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='vpp',
+                           label='Voltage peak-to-peak',
+                           get_cmd=self._meas('VPP'),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='rms',
+                           label='Voltage RMS',
+                           get_cmd=self._meas('VRMS') + ' DISPlay, DC',
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter(name='rms_no_DC',
+                           label='Voltage RMS',
+                           get_cmd=self._meas('VRMS') + ' DISPlay, AC',
+                           get_parser=float,
+                           unit='V')
+
+    def _meas(self, cmd: str) -> str:
```

Comment:
Do you want this to be a static method. Also imho the name is slightly misleading. How about `_generate_meas_str` or some such thing?